motr/m0kv/index_op.c

```diff
@@ -121,15 +144,31 @@ int index_drop(struct m0_realm *parent, struct m0_fid_arr *fids)
 	M0_PRE(fids != NULL && fids->af_count != 0);
 
 	for(i = 0; rc == 0 && i < fids->af_count; ++i) {
-		struct m0_idx   idx;
+		struct m0_idx   idx = {0};
 		struct m0_op   *op = NULL;
 
 		m0_fid_tassume(&fids->af_elems[i], &m0_dix_fid_type);
 		m0_idx_init(&idx, parent,
 			    (struct m0_uint128 *)&fids->af_elems[i]);
+
+		if (is_skip_layout) {
+			if (m0_fid_is_valid(&dix_pool_ver) && m0_fid_is_set(&dix_pool_ver))
+				set_enf_meta_flag(&idx);
+			else
+				return M0_ERR(-EINVAL);
+		} else if (m0_fid_is_set(&dix_pool_ver)) {
+			return M0_ERR(-EINVAL);
+		}
+
 		rc = m0_entity_open(&idx.in_entity, &op) ?:
-		     m0_entity_delete(&idx.in_entity, &op) ?:
-		     index_op_tail(&idx.in_entity, op, rc, NULL);
+		     m0_entity_delete(&idx.in_entity, &op);
+		if (rc != 0)
+			return M0_RC(rc);
+
+		if (is_skip_layout || is_crow_disable)
+			set_idx_flags(op);
```

Comment:
If both are disabled then _set_idx_flags()_ is not called and CROW flag is not set. I think we need to call _set_idx_flags()_ unconditionally and set all required flags in that function, we have the similar in motr code (_dix_set_idx_flags()_)

Reply:
It will work as a default behavior if set_idx_flags() was not triggered.  Default behavior CROW flag is enabled.

> motr/idx_dix.c
> static void dix_set_idx_flags(struct m0_op_idx *oi)
> {
>         if (ENABLE_DTM0)
>                 oi->oi_flags |= M0_OIF_SKIP_LAYOUT;
> 
>         if (!(oi->oi_flags & M0_OIF_SKIP_LAYOUT))
>                 oi->oi_flags |= M0_OIF_CROW;
> }
> 
> 

Above code snippet will take care it.

Reply:
Yes, but anyway that will make the code more lightweight as we will not have the same condition that is repeated 5 times.